### PR TITLE
Docker Compose and Docker Launch Configuration tests

### PR DESCRIPTION
**Pull request by jkopriva**
Docker Compose
Docker Launch Configuration

### CODE REVIEW

File: plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/ui/RunDockerImageLaunchConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+		selectTab(VOLUMES_TAB_LABEL);
+		new PushButton("Add...").click();
+		new LabeledText("Container path:").setText(containerPath);
+		new RadioButton("Mount a data volume container").click();
+		new DefaultCombo("Container").setText(containerName);
+		new OkButton().click();
+	}
+
+	// ports tab
+	public void setPublishallexposedportstorandomportsonthehostinterfaces(boolean checked) {
```
Comment (rhopp) [inline plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/ui/RunDockerImageLaunchConfiguration.java]: Is the name of this method not in camelcase by purpose?

File: plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/ui/RunDockerImageLaunchConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+		setEnableresourcelimitations(true);
+		new RadioButton("Medium").click();
+	}
+
+	public void setLow() {
+		setEnableresourcelimitations(true);
+		new RadioButton("Low").click();
+	}
+
+	public void setEnableresourcelimitations(boolean check) {
```
Comment (rhopp) [inline plugins/org.jboss.tools.docker.reddeer/src/org/jboss/tools/docker/reddeer/ui/RunDockerImageLaunchConfiguration.java]: smae question with the camelcase

File: tests/org.jboss.tools.docker.ui.bot.test/launchers/DockerAllBotTest Search Connection.launch [hunk 3]
```diff
@@ -31,7 +31,7 @@
 <stringAttribute key="org.eclipse.jdt.launching.PROGRAM_ARGUMENTS" value="-os ${target.os} -ws ${target.ws} -arch ${target.arch} -nl ${target.nl} -consoleLog"/>
 <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="org.jboss.tools.docker.ui.bot.test"/>
 <stringAttribute key="org.eclipse.jdt.launching.SOURCE_PATH_PROVIDER" value="org.eclipse.pde.ui.workbenchClasspathProvider"/>
-<stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-DsearchConnection=true -Dusage_reporting_enabled=false  -DdockerHubUsername= -DdockerHubEmail= -DdockerHubPassword="/>
+<stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-DsearchConnection=true -Dusage_reporting_enabled=false  -DdockerHubUsername= -DdockerHubEmail= -DdockerHubPassword= -DdockerComposePath=/home/jkopriva"/>
```
Comment (rhopp) [inline tests/org.jboss.tools.docker.ui.bot.test/launchers/DockerAllBotTest Search Connection.launch]: we should omit hardcoding paths in repo.

File: tests/org.jboss.tools.docker.ui.bot.test/launchers/DockerAllBotTest Socket.launch [hunk 4]
```diff
@@ -31,7 +31,7 @@
 <stringAttribute key="org.eclipse.jdt.launching.PROGRAM_ARGUMENTS" value="-os ${target.os} -ws ${target.ws} -arch ${target.arch} -nl ${target.nl} -consoleLog"/>
 <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="org.jboss.tools.docker.ui.bot.test"/>
 <stringAttribute key="org.eclipse.jdt.launching.SOURCE_PATH_PROVIDER" value="org.eclipse.pde.ui.workbenchClasspathProvider"/>
-<stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-DunixSocket=unix:///var/run/docker.sock -Dusage_reporting_enabled=false -DdockerHubUsername=${string_prompt:-DdockerHubUsername:} -DdockerHubEmail=${string_prompt:-DdockerHubEmail:} -DdockerHubPassword=${string_prompt:-DdockerHubPassword:}"/>
+<stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-DunixSocket=unix:///var/run/docker.sock -Dusage_reporting_enabled=false -DdockerHubUsername=${string_prompt:-DdockerHubUsername:} -DdockerHubEmail=${string_prompt:-DdockerHubEmail:} -DdockerHubPassword=${string_prompt:-DdockerHubPassword:} -DdockerComposePath=/home/jkopriva"/>
```
Comment (rhopp) [inline tests/org.jboss.tools.docker.ui.bot.test/launchers/DockerAllBotTest Socket.launch]: another hardcoded path

File: tests/org.jboss.tools.docker.ui.bot.test/launchers/DockerAllBotTest TCP.launch [hunk 5]
```diff
@@ -31,7 +31,7 @@
 <stringAttribute key="org.eclipse.jdt.launching.PROGRAM_ARGUMENTS" value="-os ${target.os} -ws ${target.ws} -arch ${target.arch} -nl ${target.nl} -consoleLog"/>
 <stringAttribute key="org.eclipse.jdt.launching.PROJECT_ATTR" value="org.jboss.tools.docker.ui.bot.test"/>
 <stringAttribute key="org.eclipse.jdt.launching.SOURCE_PATH_PROVIDER" value="org.eclipse.pde.ui.workbenchClasspathProvider"/>
-<stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-DdockerServerURI=http://localhost:2375 -Dusage_reporting_enabled=false  -DdockerHubUsername= -DdockerHubEmail= -DdockerHubPassword="/>
+<stringAttribute key="org.eclipse.jdt.launching.VM_ARGUMENTS" value="-DdockerServerURI=http://localhost:2375 -Dusage_reporting_enabled=false  -DdockerHubUsername= -DdockerHubEmail= -DdockerHubPassword= -DdockerComposePath=/home/jkopriva"/>
```
Comment (rhopp) [inline tests/org.jboss.tools.docker.ui.bot.test/launchers/DockerAllBotTest TCP.launch]: hardcoded path

File: tests/org.jboss.tools.docker.ui.bot.test/pom.xml [hunk 6]
```diff
@@ -11,12 +11,13 @@
...
 	<packaging>eclipse-test-plugin</packaging>
 	<properties>
-		<systemProperties>-DdockerServerURI=${dockerServerURI} -DunixSocket=${unixSocket} -DdockerHubUsername=${dockerHubUsername} -DdockerHubemail=${dockerHubEmail} -DdockerHubPassword=${dockerHubPassword}</systemProperties>
+		<systemProperties>-DdockerServerURI=${dockerServerURI} -DunixSocket=${unixSocket} -DdockerHubUsername=${dockerHubUsername} -DdockerHubemail=${dockerHubEmail} -DdockerHubPassword=${dockerHubPassword} -DdockerComposePath=${dockerComposePath}</systemProperties>
 		<unixSocket>unix:///var/run/docker.sock</unixSocket>
 		<dockerServerURI></dockerServerURI>
 		<dockerHubUsername></dockerHubUsername>
 		<dockerHubEmail></dockerHubEmail>
 		<dockerHubPassword></dockerHubPassword>
+		<dockerComposePath>/usr/local/bin</dockerComposePath>
```
Comment (rhopp) [inline tests/org.jboss.tools.docker.ui.bot.test/pom.xml]: I guess that for now it only runs on linux, right?

Comment (jkopriva) [inline tests/org.jboss.tools.docker.ui.bot.test/pom.xml]: Yes, this is default installation folder on Linux. I also plan to support Mac and Win, but not now.

Thanks for comments:)

**Comment by jkopriva** [comment]
test this

**Pull request merged**